pkg/planner/cardinality/testdata/cardinality_suite_in.json

```diff
@@ -293,6 +294,24 @@
       "explain format = 'brief' select * from t where b >= 9950 and c >= 9900 order by c limit 1"
     ]
   },
+  {
```

Comment:
This variable can affect `IndexScan -> Selection -> Limit -> IndexReader` as well (no `IndexLookUp`, no order), but it seems expected, should we add some tests for it?
```
create table t (a int, b int, c int, key(a, b, c));
insert into t values (1, 1, 1);
...
insert into t values (999,999,999);

mysql> set @@tidb_opt_ordering_index_selectivity_ratio = -1;
mysql> explain select * from t where a<1000 and c<100 limit 1;
+-------------------------------+---------+-----------+---------------------------+-------------------------------------+
| id                            | estRows | task      | access object             | operator info                       |
+-------------------------------+---------+-----------+---------------------------+-------------------------------------+
| Limit_8                       | 1.00    | root      |                           | offset:0, count:1                   |
| └─IndexReader_13              | 1.00    | root      |                           | index:Limit_12                      |
|   └─Limit_12                  | 1.00    | cop[tikv] |                           | offset:0, count:1                   |
|     └─Selection_11            | 1.00    | cop[tikv] |                           | lt(test.t.c, 100)                   |
|       └─IndexRangeScan_10     | 10.00   | cop[tikv] | table:t, index:a(a, b, c) | range:[-inf,1000), keep order:false |
+-------------------------------+---------+-----------+---------------------------+-------------------------------------+

mysql> set @@tidb_opt_ordering_index_selectivity_ratio = 0.5;
mysql> explain select * from t where a<1000 and c<100 limit 1;
+-------------------------------+---------+-----------+---------------------------+-------------------------------------+
| id                            | estRows | task      | access object             | operator info                       |
+-------------------------------+---------+-----------+---------------------------+-------------------------------------+
| Limit_8                       | 1.00    | root      |                           | offset:0, count:1                   |
| └─IndexReader_13              | 1.00    | root      |                           | index:Limit_12                      |
|   └─Limit_12                  | 1.00    | cop[tikv] |                           | offset:0, count:1                   |
|     └─Selection_11            | 22.60   | cop[tikv] |                           | lt(test.t.c, 100)                   |
|       └─IndexRangeScan_10     | 226.00  | cop[tikv] | table:t, index:a(a, b, c) | range:[-inf,1000), keep order:false |
+-------------------------------+---------+-----------+---------------------------+-------------------------------------+

mysql> set @@tidb_opt_ordering_index_selectivity_ratio = 1;
mysql> explain select * from t where a<1000 and c<100 limit 1;
+-------------------------------+---------+-----------+---------------------------+-------------------------------------+
| id                            | estRows | task      | access object             | operator info                       |
+-------------------------------+---------+-----------+---------------------------+-------------------------------------+
| Limit_8                       | 1.00    | root      |                           | offset:0, count:1                   |
| └─IndexReader_13              | 1.00    | root      |                           | index:Limit_12                      |
|   └─Limit_12                  | 1.00    | cop[tikv] |                           | offset:0, count:1                   |
|     └─Selection_11            | 90.10   | cop[tikv] |                           | lt(test.t.c, 100)                   |
|       └─IndexRangeScan_10     | 901.00  | cop[tikv] | table:t, index:a(a, b, c) | range:[-inf,1000), keep order:false |
+-------------------------------+---------+-----------+---------------------------+-------------------------------------+
```

Reply:
Please see whether to add some tests for this pattern. The rest LGTM.

Reply:
I added testcases as suggested. I also added similar testcases to TestOrderingIdxSelectivityThreshold - since this pattern also applies to that prior enhancement. If this current PR benefits from these tests - then the prior should also.